clang/test/SemaHLSL/BuiltIns/asuint-errors.hlsl

```diff
@@ -6,6 +6,10 @@ uint4 test_asuint_too_many_arg(float p0, float p1) {
   // expected-error@-1 {{no matching function for call to 'asuint'}}
   // expected-note@hlsl/hlsl_intrinsics.h:* {{candidate function template not viable: requires single argument 'V', but 2 arguments were provided}}
   // expected-note@hlsl/hlsl_intrinsics.h:* {{candidate function template not viable: requires single argument 'F', but 2 arguments were provided}}
+  // expected-note@hlsl/hlsl_intrinsics.h:* {{candidate function not viable: requires 3 arguments, but 2 were provided}}
```

Comment:
Do we need a custom note here? `candidate function not viable: requires 1 or 3 arguments, but 2 were provided`?

Reply:
This is the behavior we want. This is just the standard function overload resolution failing iterating over the possible overloads and noting them, it should cite each overload separately, not multiple overloads from a single note.